src/app/utils.ts

```diff
@@ -212,3 +211,27 @@ export const onEscHandler = (element: React.RefObject<HTMLElement>, isEnabled: b
 export const useOnEsc = (element: React.RefObject<HTMLElement>, isEnabled: boolean, cb: () => void) => {
   React.useEffect(onEscHandler(element, isEnabled, cb), [element, isEnabled]);
 };
+
+export const useOnDOMEvent = (
+  element: React.RefObject<HTMLElement>['current'] | Window ,
```

Comment:
i think if a ref is being used you'd want to pass the refObject, not the html element